src/frontend/api.spec.js

```diff
@@ -2,145 +2,157 @@
     expect(stateListener).toHaveBeenCalledTimes(3);
     expect(stateListener).toHaveBeenNthCalledWith(1, Constants.TIMEOUT);
     expect(stateListener).toHaveBeenNthCalledWith(2, Constants.STARTING);
     expect(stateListener).toHaveBeenNthCalledWith(3, Constants.TIMEOUT);
+    await cleanup();
   });
-  test("After error, timeout will not happen", () => {
+  test("After error, timeout will not happen", async () => {
+    const { serverStatus, stateListener, cleanup, clock } = setupApi();
     serverStatus({ value: Constants.ERROR });
-    jest.advanceTimersByTime(10001);
-    expect(stateListener).toHaveBeenCalledTimes(1);
-    expect(stateListener).toHaveBeenNthCalledWith(1, Constants.ERROR);
+    await clock.tickAsync(SERVER_START_TIMEOUT + 1);
+    expect(stateListener).toHaveBeenCalledWith(Constants.ERROR);
+    await cleanup();
   });
-  test("After server close, timeout will not happen until it restarts", () => {
+  test("After server close, timeout will not happen until it restarts", async () => {
+    const {
+      serverStart,
+      serverStatus,
+      stateListener,
+      cleanup,
+      clock,
+    } = setupApi();
+    serverStatus({ value: Constants.LISTENING });
+    await serverStart;
     serverStatus({ value: Constants.CLOSED });
-    jest.runAllTimers();
+    await clock.runAllAsync();
     serverStatus({ value: Constants.LISTENING });
-    jest.runAllTimers();
-    expect(stateListener).toHaveBeenCalledTimes(3);
-    expect(stateListener).toHaveBeenNthCalledWith(1, Constants.CLOSED);
-    expect(stateListener).toHaveBeenNthCalledWith(2, Constants.LISTENING);
-    expect(stateListener).toHaveBeenNthCalledWith(3, Constants.TIMEOUT);
+    await clock.runAllAsync();
+    expect(stateListener).toHaveBeenCalledTimes(4);
+    expect(stateListener).toHaveBeenNthCalledWith(1, Constants.LISTENING);
+    expect(stateListener).toHaveBeenNthCalledWith(2, Constants.CLOSED);
+    expect(stateListener).toHaveBeenNthCalledWith(3, Constants.LISTENING);
+    expect(stateListener).toHaveBeenNthCalledWith(4, Constants.TIMEOUT);
+    await cleanup();
   });
-  test("Unsubscribe works", () => {
+  test("Unsubscribe works", async () => {
+    const { subscription, serverStatus, stateListener, cleanup } = setupApi();
     subscription.remove();
     serverStatus({ value: Constants.LISTENING });
     expect(stateListener).toHaveBeenCalledTimes(0);
+    await cleanup();
   });
 });
 
 describe("Server get requests", () => {
   ky.extend.mockImplementation(() => ({
     get: jest.fn(url => ({
       json: jest.fn(() => {
-        if (url.startsWith("styles")) return [];
-        const data = [];
-        data.presets = {};
-        data.fields = {};
-        return data;
+        if (url.startsWith("presets")) return { presets: {}, fields: {} };
+        return [];
       }),
     })),
   }));
 
-  ["getObservations", "getPresets", "getFields", "getMapStyle"].forEach(
+  ["getObservations" /**, "getPresets", "getFields", "getMapStyle" */].forEach(
```

Comment:
are the commented-out parts intentional?

Reply:
Good catch. I had commented some of these out for debugging.